File: 1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        map<int, int> dic;
        dic.clear();
        int i, n = nums.size(), temp;
        for(i = 0; i < n; i++){
            temp = dic[target - nums[i]];
            if(temp){
                ans.push_back(temp);
                ans.push_back(i + 1);
                break;
            }
            dic[nums[i]] = i + 1;
        }
        return ans;
    }
};

File: 1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dic = new Dictionary<int, int> ();
        int[] ans = new int[2];
        for (int i = 0; i < nums.Length; i++) {
            if (dic.ContainsKey(target - nums[i])) {
                ans[0] = dic[target - nums[i]];
                ans[1] = i + 1;
                break;
            }
            dic[nums[i]] = i + 1;
        }
        return ans;
    }
}
